source/kotlin/src/test/kotlin/io/github/cosinekitty/astronomy/Tests.kt

```diff
@@ -732,16 +732,11 @@ class Tests {
 
     @Test
     fun `Basic search`() {
-        class CosineIdentityFunc : SearchContext {
-            override fun eval(time: Time) = time.ut - cos(time.ut)
-        }
-
         val toleranceDays = 1.0e-9
         val toleranceSeconds = toleranceDays / 86400.0
-        val func = CosineIdentityFunc()
         val time1 = Time(0.0)
         val time2 = Time(1.0)
-        val tsolve = search(func, time1, time2, toleranceSeconds)
+        val tsolve = search(time1, time2, toleranceSeconds) { time -> time.ut - cos(time.ut) }
```

Comment:
When a functional interface object is put as the ending argument Kotlin can provide this syntax and that's why it is done always at the end.

See https://kotlinlang.org/docs/fun-interfaces.html for more information.

Reply:
https://kotlinlang.org/docs/lambdas.html#passing-trailing-lambdas

"According to Kotlin convention, if the last parameter of a function is a function, then a lambda expression passed as the corresponding argument can be placed outside the parentheses:"